lxd/daemon.go

```diff
@@ -2109,6 +2109,45 @@ func (d *Daemon) nodeRefreshTask(heartbeatData *cluster.APIHeartbeat, isLeader b
 		}
 	}
 
+	// Check if a change to OVN chassis setup has occured.
```

Comment:
Please can this logic be moved into its own function, similar to `networkUpdateForkdnsServersTask(d.State(), heartbeatData)`.

As we add more refresh tasks to this function I'm thinking (although not required for this PR) that we should probably remove d.lastNodeList and split it into task-specific last-state variables. That each refresh task funtion can then take the latest `heartbeatData` check against its own tasks-specific last state var and the apply changes if needed.

Otherwise LXD can get itself into a scenario where one of the refresh tasks is failing and that will prevent the `stateChangeTaskFailure` from being set to true, which will then mean the `d.lastNodeList` is not saved with the latest data.

The end result of that scenario is that all of the refresh tasks (even the ones that previously succeeded) will then get re-applied on every heartbeat. And with this PR it means that every heartbeat the OVN networks will get restarted, which is pretty disruptive.